map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -72,8 +64,12 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
     private lateinit var buildingGroundOverlay: GroundOverlay
     private var groundOverlayGenerated: Subject<Boolean> = BehaviorSubject.createDefault(false)
     private var mapClicks: Subject<Boolean> = PublishSubject.create()
-
-    private val tour: ArticTour? get() = requireActivity().intent?.extras?.getParcelable(MapActivity.ARG_TOUR)
+    private var leaveTourDialog: AlertDialog? = null
```

Comment:
Why not use a `DialogFragment`?

Reply:
If it's going to be an `AlertDialog`, you should probably call `leaveTourDialog?.dismiss()` in `onDestroy` or `onDestroyView`, just in case the Activity has to be recreated or killed before a dialog button is pressed.

Reply:
Added `leaveTourDialog?.dismiss()` in `onDestroyView`.
Curious why you suggested `DialogFragment`?

Reply:
Just to keep it in line with the activity lifecycle. Your change covers that, so it's fine.